packages/editor/src/components/preview-dropdown/index.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * WordPress dependencies
+ */
+import { useViewportMatch } from '@wordpress/compose';
+import {
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	VisuallyHidden,
+	Icon,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { check, desktop, mobile, tablet, external } from '@wordpress/icons';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+import PostPreviewButton from '../post-preview-button';
+
+export default function PreviewDropdown( {
+	showIconLabels,
+	forceIsAutosaveable,
+} ) {
+	const { deviceType, homeUrl, isTemplate, isViewable } = useSelect(
+		( select ) => {
+			const { getDeviceType, getCurrentPostType } = select( editorStore );
+			const { getUnstableBase, getPostType } = select( coreStore );
+			const _currentPostType = getCurrentPostType();
+			return {
+				deviceType: getDeviceType(),
+				homeUrl: getUnstableBase()?.home,
+				isTemplate: _currentPostType === 'wp_template',
+				isViewable: getPostType( _currentPostType )?.viewable ?? false,
+			};
+		},
+		[]
+	);
+	const { setDeviceType } = useDispatch( editorStore );
+	const isMobile = useViewportMatch( 'medium', '<' );
+	if ( isMobile ) return null;
+
+	const popoverProps = {
+		placement: 'bottom-end',
+	};
+	const toggleProps = {
+		size: 'compact',
+		showTooltip: ! showIconLabels,
+	};
+	const menuProps = {
+		'aria-label': __( 'View options' ),
+	};
+
+	const deviceIcons = {
+		mobile,
+		tablet,
+		desktop,
+	};
+
+	return (
+		<DropdownMenu
+			className="editor-preview-dropdown"
+			popoverProps={ popoverProps }
+			toggleProps={ toggleProps }
+			menuProps={ menuProps }
+			icon={ deviceIcons[ deviceType.toLowerCase() ] }
+			label={ __( 'View' ) }
+		>
+			{ ( { onClose } ) => (
+				<>
+					<MenuGroup>
+						<MenuItem
+							onClick={ () => setDeviceType( 'Desktop' ) }
+							icon={ deviceType === 'Desktop' && check }
+						>
+							{ __( 'Desktop' ) }
+						</MenuItem>
+						<MenuItem
+							onClick={ () => setDeviceType( 'Tablet' ) }
+							icon={ deviceType === 'Tablet' && check }
+						>
+							{ __( 'Tablet' ) }
+						</MenuItem>
+						<MenuItem
+							onClick={ () => setDeviceType( 'Mobile' ) }
+							icon={ deviceType === 'Mobile' && check }
+						>
+							{ __( 'Mobile' ) }
+						</MenuItem>
+					</MenuGroup>
+					{ isTemplate && (
+						<MenuGroup>
+							<MenuItem
+								href={ homeUrl }
+								target="_blank"
+								icon={ external }
+								onClick={ onClose }
+							>
+								{ __( 'View site' ) }
```

Comment:
@jasmussen @jameskoster IMO, now that we have a "view site" link in the sidebar of the site editor, this "view site" doesn't make sense anymore and should be removed. I'm keeping for now but let me know if you agree and whether I should just get rid of it.

Reply:
Can you clarifyt which "View site" link in the sidebar you're referring to? I'm only seeing this:

![Screenshot 2023-12-11 at 08 20 32](https://github.com/WordPress/gutenberg/assets/1204802/8dc4a0ea-9778-4e31-8c62-09a810f2e165)

Do you mean this one?

![Screenshot 2023-12-11 at 08 22 07](https://github.com/WordPress/gutenberg/assets/1204802/cfea2e54-8393-4a66-9cec-2fb4e56efdc5)

I think there's an opportunity for furthering the two modes, site view and edit view, a bit more before we remove the link from the dropdown. Hopefully we can consolidate in the future, but for now this feels an ergonomic location for the link when you're in edit view.

Reply:
Ok, I'll keep it for now.